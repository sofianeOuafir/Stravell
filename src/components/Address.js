import React from "react";
import { MdMyLocation } from "react-icons/md";

const Address = ({ iconClassName, addressClassName, address }) => (
  <div className="flex align-items--center">
    <MdMyLocation className={iconClassName ? iconClassName : ''} />
    <span className={addressClassName ? addressClassName : ''}>{address}</span>
  </div>
);

export default Address;
