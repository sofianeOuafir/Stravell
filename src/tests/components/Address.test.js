import React from "react";
import { shallow } from "enzyme";
import Address from "../../components/Address";

describe("without iconClassName or addressClassName props", () => {
  test("should render correctly", () => {
    const wrapper = shallow(<Address address="Monaco" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("with iconClassName or addressClassName props", () => {
  test("should render correctly", () => {
    const wrapper = shallow(
      <Address
        iconClassName="my-css-class"
        addressClassName="my-css-class"
        address="Monaco"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
