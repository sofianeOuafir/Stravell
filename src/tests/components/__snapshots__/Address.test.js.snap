// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with iconClassName or addressClassName props should render correctly 1`] = `
<div
  className="flex align-items--center"
>
  <MdMyLocation
    className="my-css-class"
  />
  <span
    className="my-css-class"
  >
    Monaco
  </span>
</div>
`;

exports[`without iconClassName or addressClassName props should render correctly 1`] = `
<div
  className="flex align-items--center"
>
  <MdMyLocation
    className=""
  />
  <span
    className=""
  >
    Monaco
  </span>
</div>
`;
